File: app/src/main/java/io/jeffchang/dogdemo/di/modules/ActivityBuilder.java
package io.jeffchang.dogdemo.di.modules;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import io.jeffchang.dogdemo.ui.MainActivity;
import io.jeffchang.dogdemo.ui.dogdetail.DogDetailFragmentProvider;
import io.jeffchang.dogdemo.ui.doglist.DogListFragmentProvider;

/**
 * Module that exposes the main view components.
 */

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = {
            DogListFragmentProvider.class,
            DogDetailFragmentProvider.class
    })
    abstract MainActivity bindMainActivity();
}

File: app/src/main/java/io/jeffchang/buttondemo/di/modules/ActivityBuilder.kt
package io.jeffchang.buttondemo.di.modules

import dagger.Module
import dagger.android.ContributesAndroidInjector
import io.jeffchang.buttondemo.ui.MainActivity
import io.jeffchang.buttondemo.ui.usercreate.UserCreateFragmentProvider
import io.jeffchang.buttondemo.ui.userlist.UserListFragmentProvider

/**
 * Module that exposes the main view components.
 */

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = [
        UserListFragmentProvider::class,
        UserCreateFragmentProvider::class
    ])
    internal abstract fun bindMainActivity(): MainActivity
}
